frontend/packages/pipelines-plugin/src/components/pipelineruns/utils/tekton-results.ts

```diff
@@ -0,0 +1,335 @@
+import {
+  K8sResourceCommon,
+  MatchExpression,
+  MatchLabels,
+  Selector,
+} from '@console/dynamic-plugin-sdk/src';
+import { RouteModel } from '@console/internal/models';
+import { k8sGet } from '@console/internal/module/k8s';
+import { consoleProxyFetchJSON } from '@console/shared/src/utils/proxy';
+import { TektonResultModel } from '../../../models';
+import { PipelineRunKind, TaskRunKind } from '../../../types';
+
+// REST API spec
+// https://github.com/tektoncd/results/blob/main/docs/api/rest-api-spec.md
+
+// const URL_PREFIX = `/apis/results.tekton.dev/v1alpha2/parents/`;
+
+const MINIMUM_PAGE_SIZE = 5;
+const MAXIMUM_PAGE_SIZE = 10000;
+
+export type Record = {
+  name: string;
+  uid: string;
+  createTime: string;
+  updateTime: string;
+  etag: string;
+  data: {
+    // tekton.dev/v1beta1.PipelineRun | tekton.dev/v1beta1.TaskRun | results.tekton.dev/v1alpha2.Log
+    type: string;
+    value: string;
+  };
+};
+
+export type Log = {
+  result: {
+    name: string;
+    data: string;
+  };
+};
```

Comment:
Can we move these types+utils up and split it into utils, API and types maybe?

Wdyt about:

```
frontend/packages/pipelines-plugin
  /tekton-results
    /types.ts
    /utils.ts
    /api.ts
```